src/main/java/org/opentripplanner/street/model/edge/StreetEdge.java

```diff
@@ -85,6 +88,8 @@ public class StreetEdge
    */
   private float bicycleSafetyFactor;
 
+  public Map<MobilityProfile, Float> profileCost = new HashMap<>();
```

Comment:
If you notice that this causes a noticeable slowdown you can also create an array of and the use `MobilityProfile.ordinal` to access the values and saves some memory as arrays are extremely runtime- and space-efficient.

However, my gut feeling is that performance is not a huge priority so it's probably good enough as it is.

Reply:
Won't do at this time.

Reply:
@leonardehrenfried SonarLint suggests using `EnumMap` instead of `HashMap`, however, using `EnumMap` makes OTP unable to deserialize the graph. Have you seen this issue elsewhere?

Reply:
EnumMap is indeed faster than a HashMap. EnumMap also says that it is serializable.

In OTP we do use enum map sometimes but I cannot find an example where it is actually store in the graph file.

Can you post the stacktrace please?

Reply:
No need to post the stacktrace, I tried it out myself.

Serializing an enum map is indeed tricky as the default serialization uses JDK internals. However, if you want to do it anyway, here is a commit that shows you how to do it: https://github.com/ibi-group/OpenTripPlanner/commit/797d7276ee3e13f35d5bdc6c90bacc7f13a01513

Of course you need to swap out the `TransitMode` with the enum that you want to serialize.

Reply:
Thank you, that looks useful.